diff --git a/index.d.ts b/index.d.ts
index bc8a39c14bfd1d6195a39d446e895cadbc562c0e..b0a979a6ed534fc115e0f35621a40be49f3356dc 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -9,7 +9,7 @@
 import events = require('events');
 import stream = require('stream');
 import pgTypes = require('pg-types');
-import { NoticeMessage } from 'pg-protocol/dist/messages';
+import { NoticeMessage } from 'pg-protocol/dist/messages.js';
 
 import { ConnectionOptions } from 'tls';
 
