diff --git a/lib/prepare.js b/lib/prepare.js
index 232a3e4dd9a2821e66378feba0860cd885f67e1c..a71a5ff14f914d27f16ae222d78ebab844e3b34d 100644
--- a/lib/prepare.js
+++ b/lib/prepare.js
@@ -9,7 +9,7 @@ module.exports = async (npmrc, {tarballDir, pkgRoot}, {cwd, env, stdout, stderr,
 
   const versionResult = execa(
     'pnpm',
-    ['version', version, '--userconfig', npmrc, '--no-git-tag-version', '--allow-same-version'],
+    ['version', version, '--config', npmrc, '--no-git-tag-version', '--allow-same-version'],
     {
       cwd: basePath,
       env,
@@ -23,7 +23,7 @@ module.exports = async (npmrc, {tarballDir, pkgRoot}, {cwd, env, stdout, stderr,
 
   if (tarballDir) {
     logger.log('Creating npm package version %s', version);
-    const packResult = execa('pnpm', ['pack', basePath, '--userconfig', npmrc], {cwd, env, preferLocal: true});
+    const packResult = execa('pnpm', ['pack', basePath, '--config', npmrc], {cwd, env, preferLocal: true});
     packResult.stdout.pipe(stdout, {end: false});
     packResult.stderr.pipe(stderr, {end: false});
 
diff --git a/lib/publish.js b/lib/publish.js
index 3d97124963becac6535aed3ad31bb236dc3c2f51..5db565fb0b657c784ec008b8becb109f52549a93 100644
--- a/lib/publish.js
+++ b/lib/publish.js
@@ -1,5 +1,6 @@
 const path = require('path');
 const execa = require('execa');
+const fs = require('fs/promises');
 const getRegistry = require('./get-registry');
 const getChannel = require('./get-channel');
 const getReleaseInfo = require('./get-release-info');
@@ -19,10 +20,20 @@ module.exports = async (npmrc, {npmPublish, pkgRoot}, pkg, context) => {
     const registry = getRegistry(pkg, context);
     const distTag = getChannel(channel);
 
-    logger.log(`Publishing version ${version} to npm registry on dist-tag ${distTag}`);
+    logger.log(`[Kirill] Publishing version ${version} to npm registry on dist-tag ${distTag}`);
+
+    // not copying, since we aim to use oidc
+    // await fs.copyFile(
+    //   npmrc,
+    //   '.npmrc',
+    // )
+
     const result = execa(
       'pnpm',
-      ['publish', basePath, '--userconfig', npmrc, '--tag', distTag, '--registry', registry],
+      ['publish', basePath,
+        '--config', npmrc,
+        '--no-git-checks',
+        '--tag', distTag, '--registry', registry],
       {cwd, env, preferLocal: true}
     );
     result.stdout.pipe(stdout, {end: false});
diff --git a/lib/verify-auth.js b/lib/verify-auth.js
index fa2c96c30957df19148e337f785f99a2f904ea02..4690796ea9147b5eda8886f6a0ef40f06e37142c 100644
--- a/lib/verify-auth.js
+++ b/lib/verify-auth.js
@@ -18,7 +18,7 @@ module.exports = async (npmrc, pkg, context) => {
 
   if (normalizeUrl(registry) === normalizeUrl(DEFAULT_NPM_REGISTRY)) {
     try {
-      const whoamiResult = execa('pnpm', ['whoami', '--userconfig', npmrc, '--registry', registry], {
+      const whoamiResult = execa('pnpm', ['whoami', '--config', npmrc, '--registry', registry], {
         cwd,
         env,
         preferLocal: true,
